tools/mpy-tool.py

```diff
@@ -1765,6 +1765,55 @@ def copy_section(file, offset, offset2):
             f.write(merged_mpy)
 
 
+def parse_extract_segments_arg(arg):
+    kinds = set()
+    if arg is not None:
+        for kind in arg.lower().split(','):
+            if kind == 'meta':
+                kinds.add(MPYSegment.META)
+            elif kind == 'qstr':
+                kinds.add(MPYSegment.QSTR)
+            elif kind == 'obj':
+                kinds.add(MPYSegment.OBJ)
+            elif kind == 'code':
+                kinds.add(MPYSegment.CODE)
+            else:
+                raise Exception("unknown segment kind %s" % (kind,))
+    return kinds
+
+
+def extract_segments(compiled_modules, basename, kinds_arg):
+    import re
+
+    kinds = parse_extract_segments_arg(kinds_arg)
+    segments = []
+    for module in compiled_modules:
+        for segment in module.mpy_segments:
+            if not kinds or segment.kind in kinds:
```

Comment:
If you didn't want to bother validating `kinds_arg`, this could simply be `if not kinds or segment.kind in kinds_arg`.